packages/fxa-settings/src/components/TotpInputGroup/index.tsx

```diff
@@ -0,0 +1,251 @@
+    }
+  };
+  // TODO in FXA-7888 review for LTR/RTL
+  // should the inputs flow from right to left or left to right?
+  // make sure that key behaviour works as expected in RTL
+  const handleKeyDown = (
+    e: React.KeyboardEvent<HTMLInputElement>,
+    index: number
+  ) => {
+    switch (e.key) {
+      case 'Backspace':
+        e.preventDefault();
+        if (index > 0) {
+          focusOnPreviousInput(index);
+        }
+        handleBackspace(index);
+        break;
+      case 'Delete':
+        e.preventDefault();
+        if (index < codeLength) {
+          handleDelete(index);
+        }
+        break;
+      case 'ArrowRight':
+        if (index < codeLength - 1) {
+          focusOnNextInput(index);
+        }
+        break;
+      case 'ArrowLeft':
+        if (index > 0) {
+          focusOnPreviousInput(index);
+        }
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handleInput = async (
+    e: React.ChangeEvent<HTMLInputElement>,
+    index: number
+  ) => {
+    e.preventDefault();
+    errorMessage && setErrorMessage('');
+    const currentCodeArray = [...codeArray];
+
+    const value = e.target.value;
+    // allows for direct replacement of input content
+    currentCodeArray[index] = value[value.length - 1];
+
+    await setCodeArray(currentCodeArray);
+
+    if (value && index < codeLength - 1) {
+      focusOnNextInput(index);
+    } else {
+      focusOnSpecifiedInput(index);
+    }
+  };
+
+  const handlePaste = async (
+    e: React.ClipboardEvent<HTMLInputElement>,
+    index: number
+  ) => {
+    let currentIndex = index;
+    const currentCodeArray = [...codeArray];
+    const clipboardText = e.clipboardData.getData('text');
+    let array = clipboardText.split('').slice(0, codeLength - index);
+    array.forEach((character: string) => {
+      // TODO in FXA-7888 handle case where non-number value is input
+      currentCodeArray[currentIndex] = character;
+      if (currentIndex < codeLength - 1) {
+        focusOnNextInput(index);
+        currentIndex++;
+      }
+    });
+    await setCodeArray(currentCodeArray);
+    // if last pasted character is on last input, focus on that input
+    // otherwise focus on next input after last pasted character
+    focusOnSpecifiedInput(currentIndex);
+  };
+
+  const SingleDigitInput = ({ index }: SingleInputProps) => {
+    const [isFocused, setIsFocused] = useState(false);
+
+    // number used for localized message starts at 1
+    const inputNumber = index + 1;
+    return (
+      <span
+        key={`code-box-${index}`}
+        className={classNames(
+          'flex-1 max-w-18 min-h-14 w-full rounded outline-none border',
+          !errorMessage && !isFocused && 'border-grey-200 ',
+          errorMessage && 'border-red-700',
+          errorMessage && isFocused && 'shadow-input-red-focus',
+          !errorMessage &&
+            isFocused &&
+            'border-blue-400 shadow-input-blue-focus'
+        )}
```

Comment:
nit: there might be a slightly cleaner way for this using object syntax. Per https://www.npmjs.com/package/classnames#usage

```js
classnames(
  'flex-1 max-w-18 min-h-14 w-full rounded outline-none border',
  {
    'border-red-700': errorMessage,
    'shadow-input-red-focus': errorMessage && isFocused,
    'border-blue-400 shadow-input-blue-focus': !errorMessage & isFocused,
    'border-grey-200': !errorMessage && !isFocused,
  }
)
```